frontend/public/components/autocomplete.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+import { TextInput } from '@patternfly/react-core';
+import { useDocumentListener, getLabelsAsString } from '@console/shared';
+import { KeyEventModes } from '@console/shared/src/hooks';
+import { fuzzyCaseInsensitive } from './factory/table-filters';
+import { K8sResourceKind } from '../module/k8s';
+
+const MAX_SUGGESTIONS = 5;
+
+const labelParser = (resource: K8sResourceKind[]) => {
+  const labels = _.map(resource, getLabelsAsString);
+  return new Set(_.flatten(labels));
```

Comment:
You lose the benefit of using `Set` this way. The reason for `Set` is to avoid creating a giant array with duplicate labels from every resource.

```suggestion
  return resources.reduce((acc: Set<string>, resource: K8sResourceCommon) => {
    getLabelsAsString(resource).forEach((label: string) => acc.add(label));
  }, new Set<string>());
```